src/unity/python/turicreate/cython/cy_variant.pyx

```diff
@@ -694,6 +694,20 @@ cdef variant_type from_value(object v) except *:
 
 ################################################################################
 
+cdef read_archive_version(variant_map_type& d):
+    """
+    Reads only the 'archive_version' from a model state.
+    """
+    cdef variant_map_type_iterator it = d.begin()
+    ret = 0
```

Comment:
would `-1` make more sense here?

Reply:
I wasn't sure how far the convention of storing `archive_version` dates back. If there are old models that does not store it, but can still be loaded by the legacy loader, we could be unnecessarily cutting off support.

However, if it does date back I like this idea. The change needs to be coupled with handling and raising a `ToolkitError` in `_model.py` though, so I'll add that too.

Reply:
@ylow What about this? Do you think we should assume 0 if not encoded?

Reply:
0 is fine.